plugin/src/main/scala-2.12.14/ch/epfl/scala/profilers/tools/QuantitiesHijacker.scala

```diff
@@ -0,0 +1,14 @@
+package ch.epfl.scala.profilers.tools
+
+import scala.collection.mutable
+import scala.tools.nsc.Global
+import scala.reflect.internal.util.Statistics
+
+object QuantitiesHijacker {
+  type Quantities = mutable.HashMap[String, Statistics#Quantity]
+  def getRegisteredQuantities[G <: Global](global: G): Quantities = {
+    val clazz = global.statistics.getClass()
+    val field = clazz.getField("scala$reflect$internal$util$Statistics$$qs")
+    field.get(global.statistics).asInstanceOf[Quantities]
```

Comment:
### ~2.12.13
```scala
    val method = clazz.getMethod("scala$reflect$internal$util$Statistics$$qs")
    method.invoke(global.statistics).asInstanceOf[Quantities]
```

### ~2.12.14
```scala
    val field = clazz.getField("scala$reflect$internal$util$Statistics$$qs")
    field.get(global.statistics).asInstanceOf[Quantities]
```

---

`qs` became `private[this]` in this commit, and we are no longer able to access the field from the getter method. https://github.com/scala/scala/commit/301a5aec4df400e7ef8ed62f8d5e3465f9186d0a#diff-562a5d995bb7c9089b4bf89004fae2839275e8840e968f92dc4c9d1e2937a412

```sh
❯ cat Test.scala
import collection.mutable

class Test {
  val foo = "Some string"
  private val bar = "Some string"
  private[this] val baz = new mutable.HashMap[String, String]
}

$ scalac Test.scala
$ javap -private Test
Compiled from "Test.scala"
public class Test {
  private final java.lang.String foo;
  private final java.lang.String bar;
  private final scala.collection.mutable.HashMap<java.lang.String, java.lang.String> baz;
  public java.lang.String foo();
  private java.lang.String bar();
  public Test();
}
```

Reply:
Couldn't you access the field *always* with `getField`, irrespective of the scalac version? This would remove the need for version-dependent sources for this.

If not, you can also do the following:
```scala
try {
  val field = clazz.getField("scala$reflect$internal$util$Statistics$$qs")
  field.get(global.statistics).asInstanceOf[Quantities]
} catch {
  case _: NoSuchFieldException =>
    val method = clazz.getMethod("scala$reflect$internal$util$Statistics$$qs")
    method.invoke(global.statistics).asInstanceOf[Quantities]
}
```

Reply:
> Couldn't you access the field always with getField, irrespective of the scalac version?
Yeah, it sounds legit, but it doesn't work 🤔 , but your try-catch pattern works 👍